newIDE/app/src/MainFrame/index.js

```diff
@@ -304,6 +312,22 @@ class MainFrame extends React.Component<Props, State> {
 
   openFromPathOrURL = (url: string, cb: Function) => {
     const { i18n } = this.props;
+
+    const autoSavePath = url + '.autosave';
+    if (fs.existsSync(autoSavePath)) {
+      const autoSavedTime = fs.statSync(autoSavePath).mtime.getTime();
+      const saveTime = fs.statSync(url).mtime.getTime();
```

Comment:
fs.statSync can throw an exception if the path is not found, which would crash the app as it's not handled.
To fix this, wrap the appropriate code in a `try{}catch(exception){}`. 

What to do if we catch the exception? Probably nothing. This means that we have been unable to access to either the autosave or the file, but what we're doing here is a "best effort" check. If we're unable to check the autosave, then we assume there is none and don't say anything to the user.

We could imagine that this is the case if some permissions are applied and the user has no permission to read the autosave file.

Reply:
I think it should print a console.error and return false to opening the autosave